packages/circuit-ui/components/ComparisonTable/components/PlanTable/PlanTable.tsx

```diff
@@ -108,28 +107,31 @@ export const PlanTable = forwardRef<HTMLTableElement, PlanTableProps>(
         0,
       ) > COLLAPSE_THRESHOLD,
     );
-    const [headerHeight, setHeaderHeight] = useState(0);
+    const [sectionOffset, setSectionOffset] = useState(0);
     const isPlanPickerVisible = headers.length > 2;
-    const offset =
-      headerHeight +
-      (isPlanPickerVisible ? (isMobile ? 80 : 0) + (isTablet ? 16 : 0) : 0);
-
-    const updateHeaderHeight = useCallback(() => {
-      throttle(() => {
-        setHeaderHeight(theadRef.current?.getBoundingClientRect().height ?? 0);
-      }, 500)();
-    }, []);
 
     useEffect(() => {
-      updateHeaderHeight();
-    }, [updateHeaderHeight]);
+      const tableHeaderElement = theadRef.current;
+      if (!tableHeaderElement || typeof ResizeObserver === 'undefined') {
+        return undefined;
+      }
 
-    useEffect(() => {
-      window.addEventListener('resize', updateHeaderHeight);
+      // opt for progressive enhancement
+      // eslint-disable-next-line compat/compat
+      const headerSizeObserver = new ResizeObserver((entries) => {
+        const planPickerHeight = isPlanPickerVisible
+          ? (isMobile ? 80 : 0) + (isTablet ? 16 : 0)
+          : 0;
+        setSectionOffset(entries[0].contentRect.height + planPickerHeight);
+      });
+
+      headerSizeObserver.observe(tableHeaderElement);
       return () => {
-        window.removeEventListener('resize', updateHeaderHeight);
+        if (tableHeaderElement) {
+          headerSizeObserver.unobserve(tableHeaderElement);
+        }
```

Comment:
This check is now obsolete since `tableHeaderElement` is guaranteed to be defined by the check on line 115. 

```suggestion
        headerSizeObserver.unobserve(tableHeaderElement);
```